db/migrate/20200912032913_add_ancestry_to_resource_pool.rb

```diff
@@ -0,0 +1,130 @@
+class AddAncestryToResourcePool < ActiveRecord::Migration[5.2]
+  class ResourcePool < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    has_many :all_relationships, :class_name => "AddAncestryToResourcePool::Relationship", :dependent => :destroy, :as => :resource
+    include ActiveRecord::IdRegions
+  end
+
+  class Relationship < ActiveRecord::Base
+    belongs_to :resource_pool, :class_name => 'AddAncestryToResourcePool::ResourcePool', :foreign_key => :resource_pool_id
+  end
+
+  def up
+    add_column :resource_pools, :ancestry, :string
+    add_index :resource_pools, :ancestry
+
+    say_with_time("set vm ancestry from existing genealogy relationship resource information") do
+      transfer_relationships_to_ancestry(ResourcePool, 'ResourcePool', 'genealogy')
+    end
+
+    Relationship.where(:relationship => 'genealogy', :resource_type => 'ResourcePool', :resource_id => ResourcePool.all.select(:id)).delete_all
```

Comment:
too bad we duplicated this query. once in sql (with that lateral join stuff) and once in active record here.

Reply:
well, in a perfect world we have one more iteration of this code for folders, so if you're invested in me redoing for that, that's certainly your prerogative.